File: app/src/main/java/cat/teknos/berry/view/activity/MainActivity.java
package cat.teknos.berry.view.activity;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Button;
import android.widget.EditText;
import android.widget.RadioButton;

import cat.teknos.berry.R;

public class MainActivity extends AppCompatActivity {

    private EditText etName;
    private RadioButton rbEasy, rbHard;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        etName = findViewById(R.id.etName);
        rbEasy = findViewById(R.id.rbEasy);
        rbHard = findViewById(R.id.rbHard);

        Button btnPlay = findViewById(R.id.btnPlay);
        btnPlay.setOnClickListener(v -> play());
    }

    public void play() {
        int level = 2;
        if (rbEasy.isChecked()) {
            level = 1;
        } else if (rbHard.isChecked()) {
            level = 3;
        }

        String playerName = etName.getText().toString();
        if (playerName.isEmpty()) {
            playerName = "PLAYER";
        }

        Intent intent = new Intent(this, GameActivity.class);
        intent.putExtra("level", level);
        intent.putExtra("playerName", playerName);
        startActivity(intent);
    }
}

File: app/src/main/java/cat/teknos/berry/view/activity/MainActivity.kt
package cat.teknos.berry.view.activity

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.RadioButton
import androidx.appcompat.app.AppCompatActivity
import cat.teknos.berry.R

class MainActivity : AppCompatActivity() {
    private var etName: EditText? = null
    private var rbEasy: RadioButton? = null
    private var rbHard: RadioButton? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        etName = findViewById(R.id.etName)
        rbEasy = findViewById(R.id.rbEasy)
        rbHard = findViewById(R.id.rbHard)

        val btnPlay = findViewById<Button>(R.id.btnPlay)
        btnPlay.setOnClickListener { _: View? -> play() }
    }

    private fun play() {
        val levelNumber: Int
        val levelName: String

        when {
            rbEasy!!.isChecked -> {
                levelNumber = 1
                levelName = "EASY"
            }
            rbHard!!.isChecked -> {
                levelNumber = 3
                levelName = "HARD"
            }
            else -> {
                levelNumber = 2
                levelName = "MEDIUM"
            }
        }

        var playerName = etName!!.text.toString()
        if (playerName.isEmpty()) {
            playerName = "PLAYER"
        }

        val intent = Intent(this, GameActivity::class.java)
        intent.putExtra("levelNumber", levelNumber)
        intent.putExtra("levelName", levelName)
        intent.putExtra("playerName", playerName)
        startActivity(intent)
    }
}